src/lib/ast.rs

```diff
@@ -0,0 +1,623 @@
+// Copyright (c) 2017 King's College London
```

Comment:
Do we need to have a license on every file? I tend to put one license file in the repo top-level, then you don't need to maintain the copyright year (missing in this file) in multiple places. Not sure what Laurie usually does, perhaps look at his lrtable repo?